ledger/src/shred.rs

```diff
@@ -532,9 +532,7 @@ pub mod layout {
 
     pub fn get_shred(packet: &Packet) -> Option<&[u8]> {
         let size = get_shred_size(packet)?;
-        let shred = packet.data(..size)?;
-        // Should at least have a signature.
-        (size >= SIZE_OF_SIGNATURE).then(|| shred)
```

Comment:
all subsequent `shred::layout` method do verify size before offsetting into the buffer.